src/benchmarklib/benchmark_utils.cpp

```diff
@@ -68,4 +77,140 @@ BenchmarkConfig::BenchmarkConfig(const BenchmarkMode benchmark_mode, const bool
       enable_visualization(enable_visualization),
       out(out) {}
 
+BenchmarkConfig BenchmarkConfig::get_default_config() { return BenchmarkConfig(); }
+
+bool CLIConfigParser::cli_has_json_config(const int argc, char** argv) {
+  const auto has_json = argc > 1 && boost::algorithm::ends_with(argv[1], ".json");
+  if (has_json && argc > 2) {
+      std::cout << "Passed multiple args with a json config. All CLI args will be ignored...";
+  }
+
+  return has_json;
+}
+
+nlohmann::json CLIConfigParser::config_file_to_json(const std::string& json_file_str) {
+  Assert(filesystem::is_regular_file(json_file_str), "No such file " + json_file_str);
+
+  nlohmann::json json_config;
+  std::ifstream json_file{json_file_str};
+  json_file >> json_config;
+
+  return json_config;
+}
+
+BenchmarkConfig CLIConfigParser::parse_default_json_config(const nlohmann::json& json_config) {
+  const auto default_config = BenchmarkConfig::get_default_config();
+
+  // Should the benchmark be run in verbose mode
+  const auto verbose = json_config.value("verbose", default_config.verbose);
+  auto& out = get_out_stream(verbose);
+
+  // In non-verbose mode, disable performance warnings
+  std::optional<PerformanceWarningDisabler> performance_warning_disabler;
+  if (!verbose) {
+    performance_warning_disabler.emplace();
+  }
+
+  // Display info about output destination
+  std::optional<std::string> output_file_path;
+  const auto output_file_string = json_config.value("output", "");
+  if (!output_file_string.empty()) {
+    output_file_path = output_file_string;
+    out << "- Writing benchmark results to '" << *output_file_path << "'" << std::endl;
+  } else {
+    out << "- Writing benchmark results to stdout" << std::endl;
+  }
+
+  // Display info about MVCC being enabled or not
+  const auto enable_mvcc = json_config.value("mvcc", default_config.use_mvcc == UseMvcc::Yes);
+  const auto use_mvcc = enable_mvcc ? UseMvcc::Yes : UseMvcc::No;
+  out << "- MVCC is " << (enable_mvcc ? "enabled" : "disabled") << std::endl;
+
+  // Initialise the scheduler if the benchmark was requested to run multi-threaded
```

Comment:
Naming: Initializing the scheduler goes way beyond just parsing the json object.